metricbeat/module/system/raid/blockinfo/parser.go

```diff
@@ -0,0 +1,207 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package blockinfo
+
+import (
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/elastic/beats/libbeat/common"
+)
+
+//get the raid level and use that to determine how we fill out the array
+//Only data-reduntant RIAD levels (1,4,5,6,10) have some of these fields
+func isRedundant(path string) (bool, error) {
+	raw, err := ioutil.ReadFile(filepath.Join(path, "md", "level"))
+	if err != nil {
+		return false, err
+	}
+	raidStr := strings.TrimSpace(string(raw))
+	if raidStr == "raid1" || raidStr == "raid4" || raidStr == "raid5" ||
+		raidStr == "raid6" || raidStr == "raid10" {
+		return true, nil
+	}
+
+	return false, nil
+}
+
+func parseIntVal(path string) (int64, error) {
+	var value int64
+	raw, err := ioutil.ReadFile(path)
+	if err != nil {
+		return value, err
+	}
+	strVal := strings.TrimSpace(string(raw))
+
+	value, err = strconv.ParseInt(string(strVal), 10, 64)
+	if err != nil {
+		return value, err
+	}
+
+	return value, nil
+}
+
+//get the current sync status as it exists under md/sync_completed
+//if there's no sync operation in progress, the file will just have 'none'
+//in which case, default to to the overall size
+func getSyncStatus(path string, size int64) (SyncStatus, error) {
+
+	raw, err := ioutil.ReadFile(filepath.Join(path, "md", "sync_completed"))
+	if err != nil {
+		return SyncStatus{}, errors.Wrap(err, "could not open sync_completed")
+	}
+	if string(raw) == "none\n" {
```

Comment:
As we later we also need it maybe it'd be good to trim the string already here to avoid surprises with these comparisons.